crates/factor-sqlite/src/runtime_config/spin.rs

```diff
@@ -0,0 +1,237 @@
+//! Spin's default handling of the runtime configuration for SQLite databases.
+
+use std::{
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use serde::Deserialize;
+use spin_factors::anyhow::{self, Context as _};
+use spin_world::v2::sqlite as v2;
+use tokio::sync::OnceCell;
+
+use crate::{Connection, ConnectionPool, SimpleConnectionPool};
+
+use super::RuntimeConfigResolver;
+
+/// Spin's default handling of the runtime configuration for SQLite databases.
+///
+/// This type implements the [`RuntimeConfigResolver`] trait and provides a way to
+/// opt into the default behavior of Spin's SQLite database handling.
+pub struct SpinSqliteRuntimeConfig {
+    state_dir: PathBuf,
+    base_path: Option<PathBuf>,
+}
+
+impl SpinSqliteRuntimeConfig {
+    /// Create a new `SpinSqliteRuntimeConfig`
+    ///
+    /// This takes as arguments:
+    /// * the state directory path (i.e., the path to the `.spin` file). This
+    /// is used to derive the default path a local SQLite database file.
+    /// * the base path from which relative paths referenced in configuration are resolved
```

Comment:
I was surprised to see `.parent()` below and had to come back and reread this; I think it would be less surprising if the caller was responsible for `.parent` instead.